packages/flutter/test/widgets/default_colors_test.dart

```diff
@@ -0,0 +1,153 @@
+        const Offset(250, 50): const Color(0xFFABCDEF), // the text (above the p)
+        const Offset(250, 95): const Color(0xFFFFFFFF), // the text (the p)
+        const Offset(400, 400): const Color(0xFFABCDEF), // the background
+        const Offset(799, 599): const Color(0xFFABCDEF), // the background
+      },
+    );
+  }, skip: !canCaptureImage);
+
+  testWidgets('Default text selection color', (WidgetTester tester) async {
+    final GlobalKey key = GlobalKey();
+    await tester.pumpWidget(
+      ColoredBox(
+        color: const Color(0xFFFFFFFF),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: MediaQuery(
+            data: const MediaQueryData(),
+            child: Overlay(
+              initialEntries: <OverlayEntry>[
+                OverlayEntry(
+                  builder: (BuildContext context) => SelectableRegion(
+                    focusNode: FocusNode(),
+                    selectionControls: emptyTextSelectionControls,
+                    child: Align(
+                      key: key,
+                      alignment: Alignment.topLeft,
+                      child: const Text('Éxp', textDirection: TextDirection.ltr, style: TextStyle(fontSize: crispText)),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+    await expectColors(
+      tester,
+      find.byType(Align),
+      <Color>{ const Color(0xFFFFFFFF) },
+    );
+    // fake a "select all" event to selecte the text
+    Actions.invoke(key.currentContext!, const SelectAllTextIntent(SelectionChangedCause.keyboard));
+    await tester.pump();
+    await expectColors(
+      tester,
+      find.byType(Align),
+      <Color>{ const Color(0xFFFFFFFF), const Color(0xFFBFBFBF) }, // 0x80808080 blended with 0xFFFFFFFF
+      <Offset, Color>{
+        Offset.zero: const Color(0xFFBFBFBF), // the selected text
+        const Offset(10, 10): const Color(0xFFBFBFBF), // the selected text
+        const Offset(50, 95): const Color(0xFFBFBFBF), // the selected background (under the É)
+        const Offset(250, 50): const Color(0xFFBFBFBF), // the selected background (above the p)
+        const Offset(250, 95): const Color(0xFFBFBFBF), // the selected text (the p)
+        const Offset(400, 400): const Color(0xFFFFFFFF), // the background
+        const Offset(799, 599): const Color(0xFFFFFFFF), // the background
+      },
+    );
+  }, skip: !canCaptureImage);
+}
+
+Color getPixel(ByteData bytes, int x, int y, int width) {
+  final int offset = (x + y * width) * 4;
+  return Color.fromARGB(
+    bytes.getUint8(offset + 3),
+    bytes.getUint8(offset + 0),
+    bytes.getUint8(offset + 1),
+    bytes.getUint8(offset + 2),
+  );
+}
+
+Future<void> expectColors(WidgetTester tester, Finder finder, Set<Color> allowedColors, [ Map<Offset, Color>? spotChecks ]) async {
```

Comment:
I am usually intrigued to know why are `expectColors`, `getPixel`, `crispText` and `expText` public? Are they shared elsewhere?

If they are not shared elsewhere, wouldn't we benefit from making them private? Since this way we would:
- Implicitly tell the reader this is only intended to use in `default_colors_test.dart`
- Make the test file self contained
- Avoid accidentally importing it in other test files

If they are intended to be shared elsewhere, as a reader I would appreciate some documentation over it.

Reply:
Seems reasonable.

Reply:
fixed